src/cmd/linuxkit/pkgsrc/docker.go

```diff
@@ -0,0 +1,90 @@
+package pkgsrc
+
+// Thin wrappers around Docker CLI invocations
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+const debugDockerCommands = false
+
+const dctEnableEnv = "DOCKER_CONTENT_TRUST=1"
+
+type dockerRunner struct {
+	dct   bool
+	cache bool
+}
+
+func newDockerRunner(dct, cache bool) dockerRunner {
+	return dockerRunner{dct: dct, cache: cache}
+}
+
+func (dr dockerRunner) command(args ...string) error {
+	cmd := exec.Command("docker", args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+	if dr.dct {
+		cmd.Env = append(cmd.Env, dctEnableEnv)
+	}
+	if debugDockerCommands {
+		var dct string
+		if dr.dct {
+			dct = " " + dctEnableEnv
+		}
+		fmt.Fprintf(os.Stderr, "+ %s%v\n", dct, cmd.Args)
+	}
+	return cmd.Run()
+}
+
+func (dr dockerRunner) pull(img string) (bool, error) {
+	err := dr.command("pull", img)
+	if err == nil {
+		return true, nil
+	}
+	switch err.(type) {
+	case *exec.ExitError:
+		return false, nil
+	default:
+		return false, err
+	}
+}
+
+func (dr dockerRunner) push(img string) error {
+	return dr.command("push", img)
+}
+
+func (dr dockerRunner) pushWithManifest(img, suffix string) error {
+	if err := dr.push(img + suffix); err != nil {
+		return err
+	}
+
+	dctArg := "0"
+	if dr.dct {
+		dctArg = "1"
+	}
+
+	cmd := exec.Command("linuxkit-push-manifest", img, dctArg)
```

Comment:
and here, as someone could easily not have this installed

Reply:
BTW I debated doing something really gross here to save installing a new tool, until the script could be reimplemented in go. Namely:

```
pushManifest := `
«108 lines of scripts/push-manifest.sh»
`
....
cmd := exec.Command("/bin/sh", "-c", pushManifest, «other args»)
```

Reply:
yeah its not too terrible. You could use `go generate` to keep the original in a seperate file but it hardly seems wirth it

Reply:
`go generate` would be nice because then syntax highlighting would work for anyone editing it. I think I will go this route then.